src/getAllComponents.js

```diff
@@ -0,0 +1,12 @@
+export default function getAllComponents(components) {
+  const arr = Array.isArray(components) ? components : [components];
+  const result = [];
+  for (let i = 0, l = arr.length; i < l; i++) {
```

Comment:
Could we change this to `.forEach`? Would make the code a little easier to grasp in my opinion.